### PR TITLE
React admin with git submodule

### PR DESCRIPTION
**Pull request by EmmanuelObua**

### CODE REVIEW

**Comment by twschiller** [comment]
Thanks @EmmanuelObua. This has the same problem as the last one where if you visit it without the slash https://playground-5lhd1uc4k-pixiebrix.vercel.app/react-admin the page doesn't load

Could you introduce the same fix the you and @fregante added for the vue one?

**Comment by EmmanuelObua** [comment]
> Thanks @EmmanuelObua. This has the same problem as the last one where if you visit it without the slash https://playground-5lhd1uc4k-pixiebrix.vercel.app/react-admin the page doesn't load
> 
> If you visit https://playground-5lhd1uc4k-pixiebrix.vercel.app and then click react-admin it won't load
> 
> Could you introduce the same fix the you and @fregante added for the vue demo?

I noticed this, it's failing to read the manifest.json file, I looked around and I am suspecting it's because of the way the manifest file is linked in the build index.html file with ./manifest.json syntax.

I wanted to try again by changing the way it's link in the href from ./manifest.json to manifest.json

**Comment by twschiller** [comment]
The CSS and JS file are also not loading. 

image

I think you need to set the PUBLIC_URL during the build? See the build steps for the create-react-app example.

This is the index.html file of the react-admin demo. Notice that it refers to the PUBLIC_URL: https://github.com/marmelab/react-admin/blob/master/examples/demo/public/index.html#L10

**Comment by EmmanuelObua** [comment]
Ok please, sounds good

**Review by twschiller — CHANGES_REQUESTED** [review]
@EmmanuelObua this is still not working for me when I visit without the "/" (which is the what is linked on the homepage)

image

**Comment by twschiller** [comment]
Merging this in to see if it unblocks the other build

File: react-admin/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#! /bin/env bash
+
+set -e # exit when any command fails
+
+PUBLIC_URL=/react-admin/
```
Comment (fregante) [inline react-admin/build.sh:5]: When declaring variables like this, they will only be available in the current file.


To make it available in the scripts, either use:

```
PUBLIC_URL=/react-admin/ make build
```

or 


```
export PUBLIC_URL=/react-admin/

make build
```

The `export` keyword makes it available for all the sub-processes (like `make` and `npm`)


---

I already made this change on `main` branch

File: vercel.json [hunk 2]
```diff
@@ -6,6 +6,10 @@
 		{
 			"source": "/create-react-app/(.*)",
 			"destination": "/create-react-app/index.html"
+		},
+		{
+			"source": "/react-admin/(.*)",
+			"destination": "/react-admin/index.html"
```
Comment (fregante) [inline vercel.json:12]: This is not necessary when the user uses `#/` in its URLs. It can be removed

**Review by fregante — COMMENTED** [review]
It works now! https://playground-gjhxthfrs-pixiebrix.vercel.app/react-admin#/

Here's a couple of notes for the future reference

**Pull request merged**